Payment/Factories/PaymentProcessorFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Payment.Factories.Common;
+
+namespace Payment.Factories;
+
+public class PaymentProcessorFactory
+{
+    private readonly Dictionary<string, Func<decimal, decimal, PaymentProcessor>> _paymentProcessors = new();
+
+    public PaymentProcessorFactory(bool addBaseFactories = true)
+    {
+        _paymentProcessors.Add("Null", (_, _) => new CreditPaymentProcessor(0, 0));
+        if (!addBaseFactories)
```

Comment:
If I pass the false as addBaseFactories in parameter, then this code returns. am i right?

Reply:
Yes, As you mentioned in previous practice, the user must be noticed about base factories so he can decide if needs the base factories or not.
The null provider is exception because in the readme has been mentioned null provider is "else" case.